_P222_i2c-AmbiMate_MS4_.ino

```diff
@@ -0,0 +1,299 @@
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[2], PSTR(PLUGIN_VALUENAME3_222));
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[3], PSTR(PLUGIN_VALUENAME4_222));
+        //strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[4], PSTR(PLUGIN_VALUENAME5_222));
+        //strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[5], PSTR(PLUGIN_VALUENAME6_222));
+        //strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[6], PSTR(PLUGIN_VALUENAME7_222));
+        //strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[7], PSTR(PLUGIN_VALUENAME8_222));
+        break;
+      }
+
+    case PLUGIN_INIT:
+      {
+        // Data and basic information are acquired from the module
+
+        _i2caddrP222 = AMBIMATESENSOR_DEFAULT_ADDR;
+
+        // Read Firmware Version
+        unsigned char fw_ver = I2C_read8_reg(_i2caddrP222, AMBIMATESENSOR_GET_VERSION, &good);
+        // Read Firmware Subversion
+        unsigned char fw_sub_ver = I2C_read8_reg(_i2caddrP222, AMBIMATESENSOR_GET_SUBVERSION, &good);
+
+        String log = F("AmbiMate: Firmware Version: ");
+        log += String(fw_ver);
+        log += F(" Subversion: ");
+        log += String(fw_sub_ver);
+        addLog(LOG_LEVEL_INFO, log);
+
+        // Read Optional Sensors byte
+        opt_sensors = I2C_read8_reg(_i2caddrP222, AMBIMATESENSOR_GET_OPT_SENSORS, &good);
+
+        success = true;
+        break;
+      }
+
+    case PLUGIN_WEBFORM_LOAD:
+      {
+
+        addFormSeparator(2);
+
+        addFormCheckBox(F("Use Celsius"), F("p222_Celsius"), PCONFIG(0));
+
+        addFormSeparator(2);
+
+        success = true;
+        break;
+      }
+
+    case PLUGIN_WEBFORM_SAVE:
+      {
+
+        PCONFIG(0) = isFormItemChecked(F("p222_Celsius"));
+
+        success = true;
+        break;
+      }
+
+    case PLUGIN_ONCE_A_SECOND: //For motion event processing
+      {
+        uint8_t statusByte;
+        good = I2C_write8_reg(_i2caddrP222, AMBIMATESENSOR_SET_SCAN_START_BYTE, AMBIMATESENSOR_READ_PIR_ONLY);
+        delayBackground(100);
+        statusByte = I2C_read8_reg(_i2caddrP222, AMBIMATESENSOR_GET_STATUS, &good);
+
+        if (statusByte & 0x80){ // PIR Event occurred
+          UserVar[event->BaseVarIndex + 3] = 1;
+          String log = F("AmbiMate: PIR_MOTION_EVENT");
+          addLog(LOG_LEVEL_INFO, log);
+          sendData(event);
+        }
+        else{
+          UserVar[event->BaseVarIndex + 3] = 0;
+        }
+
+        success = true;
+        break;
+
+      }
+    case PLUGIN_READ:
+      {
+        unsigned char buf[20];
+
+        if (opt_sensors & 0x01) //Gas Sensor installed
+        {
+          // Read All Sensors next byte
+          good = I2C_write8_reg(_i2caddrP222, AMBIMATESENSOR_SET_SCAN_START_BYTE, AMBIMATESENSOR_READ_INCLUDE_GAS);
+        }
+        else
+        {
+          good = I2C_write8_reg(_i2caddrP222, AMBIMATESENSOR_SET_SCAN_START_BYTE, AMBIMATESENSOR_READ_EXCLUDE_GAS);
+        }
+
+        // 100ms delay
+        delayBackground(100);
```

Comment:
Why a delay here?
See the code of BME280 to get an idea of letting the plugin reschedule itself to be read when the sensor should be ready to give the processed values.
Delay will block the ESP to do other tasks.